scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +656,106 @@ def __mul__(self, other):
+    ----------
+    t : array_like, shape (N,)
+        An array of strictly increasing timestamps for the fixed rotations. The
+        number of times `N` must be equal to the number of rotations specified
+        in the `rotations` object.
+    r : a `Rotation` instance
+        This object contains the fixed rotations between which the
+        interpolation will be done. Must contain `N`rotations and at least 2.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+    """
+    def __init__(self, t, r):
+        self._key_quats = r._quat
+        self._num_rots = r._quat.shape[0]
+        self._times = np.asarray(t)
+
+        if self._num_rots == 1:
+            raise ValueError("Expected at least 2 rotations in rotation object"
+                             ", got {}.".format(self._num_rots))
+
+        if self._times.ndim > 1:
+            raise ValueError("Expected keyframe timestamp array to be 1 "
+                             "dimensional, got {} dimensions.".format(
+                                self._times.ndim))
+
+        if self._times.shape[0] != self._num_rots:
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of keyframe timestamps, got {} rotations "
+                             "and {} timestamps.".format(
+                                self._num_rots,
+                                self._times.shape[0],
+                             ))
+
+        # Make a copy for modification
+        q0 = self._key_quats[0:-1].copy()
+        q0[:, -1] *= -1
+        q1 = self._key_quats[1:]
+        self._interp_rotvecs = Rotation.from_quaternion(
+            _compose_quat(q0, q1)).as_rotvec()
+
+    def __call__(self, timestamps):
+        """Generate interpolated rotations at given times.
+
+        Given an array of `timestamps` returns a `Rotation` instance with the
+        interpolated rotations at the given times.
+
+        Parameters
+        ----------
+        timestamps : array_like, 1D
+            An array of increasing timestamps, for the interpolated rotations
+        """
+        frames = np.asarray(timestamps)
+
+        if frames.ndim > 1:
+            raise ValueError("Expected given timestamps to be 1 dimensional ,"
+                             "got {} dimensions.".format(frames.ndim))
+
+        # Index of largest keyframe time less than or equal to given time
+        # Given timestamps need not be sorted anymore
+        q0_ind = np.searchsorted(self._times, frames, side='right') - 1
+        if(np.any(q0_ind < 0) or np.any(q0_ind >= self._num_rots - 1)):
+            raise ValueError("Expected given timestamp to be within range of "
+                             "keyframe timestamps [{}, {}].".format(
+                                self._times[0], self._times[-1]))
+
+        q0 = self._key_quats[q0_ind]
```

Comment:
I don't know why, but it all reads very awkward. Try different names, with less `_` for example.

Like I don't think it will be less readable if you name it like `q0_ind` -> `ind`, `q0_q1_rotvec -> rotvec`, etc.